docs/PR_GUIDE.md

```diff
@@ -0,0 +1,92 @@
+# Pull Request Guide
+
+## General development rules:
+
+- All the changes of the code base are implemented through pull requests (PR) on master or other branches
+- When all the features for an upcoming release are implemented and are stable enough, a new release branch 
+  is made from master. After that, all app versions are updated in the master branch in files:
+  ./omim/xcode/common.xcconfig and ./omim/android/gradle.properties
+- Only PR with bugfix should be made on release branches. If it’s necessary it’s permitted to cherry-pick such
+  commits to master
+- If a commit is made on a release branch and it is not intended to be merged to master the text of the commit
+  should have prefix [RELEASE ONLY]
+- When a release is published, one of developers makes a PR with all new commits of the release to master
+- All the further commits to release branch should be cherry-picked to the master by commits author
+- If commits are mistakenly merged into master or release branch, a PR with new commits which revert them should be made
+- When a PR to any branch is ready to merge, rebase and merge option should be used to make the code source tree
+  flatter
+
+
+## Writing good PR:
+
+- PR should be small and reflect only one idea, a feature, a bugfix, or a refactoring. In most cases a PR of 100 lines
+  or less is considered as a PR of normal size and a PR of 1000 lines as big one
+- If a PR implements two different features, refactoring, or bugfixes, it should be split into several PRs
+- The description field of every PR should contain a description, links to a ticket or/and links to the confluence
+- If a PR implements a complex math task, the description should contain a link to the proof of the solution or a link
+  to a trusted source
+- All PRs should contain a link to the corresponding ticket. The exception is refactoring if there’s nothing to test
+  after it. The reason for that is that every feature or bugfix should be tested by testers after it’s implemented
+- Every commit of all PRs should be compilable under all platforms. All tests should pass. So if changing of code
+  breaks unit or integration tests these tests should be fixed in the same commit
+- Every commit should reflect a completed idea and have an understandable comment. Review fixes should be merged into
+  one commit
+- New functionality and unit or integration tests for it should be developed in the same PR
+- When some source files are changed and then some other source files based on it are auto generated, they should be
+  committed in different commits. For example, one commit changes strings.txt and another one contains generated files
+- Most commits and PRs should have prefixes in square brackets depending on the changed subsystem. For example,
```

Comment:
s/A commits and a PRs/Commits and PRs/

Reply:
Fixed: `Commits and PRs may have several prefixes`, but it's from another part of text.